sync15-adapter/src/bso_record.rs

```diff
@@ -206,6 +207,44 @@ impl Payload {
         serde_json::to_string(&JsonValue::from(self))
             .expect("JSON.stringify failed, which shouldn't be possible")
     }
+
+    /// "Auto" fields are fields like 'sortindex' (and potentially 'ttl' in
+    /// the future) which are:
+    ///
+    /// - Added to the payload automatically when deserializing if present on
+    ///   the incoming BSO.
+    /// - Removed from the payload automatically and attached to the BSO if
+    ///   present on the outgoing payload.
+    pub(crate) fn add_auto_field<T: Into<JsonValue>>(&mut self, name: &str, v: Option<T>) {
```

Comment:
We don't call this outside this module so this doesn't need to be `pub` or `pub(crate)`.